### PR TITLE
Fix wallet balance inconsistency issue

### PR DESCRIPTION
**Pull request by rlan35**

### CODE REVIEW

**Comment by alajko** [comment]
It trusts the first node more than other nodes.

File: cmd/client/wallet/main.go [hunk 1]
```diff
@@ -563,34 +571,46 @@ func convertBalanceIntoReadableFormat(balance *big.Int) string {
...
+				response, err := client.GetBalance(address)
+				if err != nil {
+					log.Info("failed to get balance, retrying ...")
+					time.Sleep(200 * time.Millisecond)
+					continue
+				}
+				log.Debug("FetchBalance", "response", response)
+				respBalance := big.NewInt(0)
+				respBalance.SetBytes(response.Balance)
+				if balance.Cmp(respBalance) < 0 {
```
Comment (chaosma) [inline cmd/client/wallet/main.go]: what if some server is slower, and the balance is greater than previous but not the latest? should we loop through the end of the loop and pick the maximum value as latest balance

Comment (alajko) [reply cmd/client/wallet/main.go]: I don't think picking maximum balance is correct. Maximum nonce, might be more correct to reflect more recent transactions.

**Comment by alajko** [comment]
We just trust the first node as vs trusting a randomn node. So I don't understand this change.

File: cmd/client/wallet/main.go (hunk 1 above)
Comment (chaosma) [reply cmd/client/wallet/main.go]: right, maximum nonce

Comment (alajko) [reply cmd/client/wallet/main.go]: Maximum value might not be the correct state as well.

Comment (rlan35) [reply cmd/client/wallet/main.go]: yea, we have to rely on consistent nodes to check. This is already working fine. No major problem so far.

Comment (alajko) [reply cmd/client/wallet/main.go]: Ideally the correct way is picking max of (nonce, balance). (In that order) per shard.

**Pull request merged**